tasks/test_results_processor.py

```diff
@@ -161,11 +161,12 @@ def process_individual_arg(self, upload: Upload, repository) -> List[Testrun]:
 
         testrun_list = []
 
-        for file in data["test_results_files"]:
-            file = file["data"]
+        for file_dict in data["test_results_files"]:
+            filename = file_dict["filename"]
+            file = file_dict["data"]
```

Comment:
What's your POV on dataclasses vs dicts?

Reply:
we're loading this from json and by default `json.loads` returns a dict, but it might be worth looking for a solution where it's loaded into a class

Reply:
:bufo-ack:

Yeah I like dataclasses cause it's nice for the IDE for autocomplete, plus we know what the response is, but yea for the future